src/Microsoft.VisualStudio.Editors/SettingsDesigner/DesignTimeSettings.vb

```diff
@@ -177,15 +177,15 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
         ''' <returns></returns>
         ''' <remarks></remarks>
         Friend Shared Function EqualIdentifiers(Id1 As String, Id2 As String) As Boolean
-            Return String.Equals(Id1, Id2, StringComparison.OrdinalIgnoreCase)
+            Return StringComparers.SettingNames.Equals(Id1, Id2)
         End Function
 
         Friend Function CreateUniqueName(Optional Base As String = Nothing) As String
             If String.IsNullOrEmpty(Base) Then
                 Base = My.Resources.Microsoft_VisualStudio_Editors_Designer.SD_DefaultSettingName
             End If
 
-            Dim ExistingNames As New Hashtable
+            Dim ExistingNames As New Hashtable(StringComparers.SettingNames)
```

Comment:
It would be nice to ban `New Dictionary(of String, T)()` and force specifying a comparer.

Reply:
Not sure how it handles specifically instantiated generics.  And of course, this is just a `Hashtable` that you are putting strings in, so even harder to find...